src/async_cache.rs

```diff
@@ -41,7 +41,7 @@ impl<Key: Eq + Hash + Send + Clone, Value: Debug + Clone + Send, Error: Debug +
         &self,
         key: Key,
         or_else: impl FnOnce() -> Pin<Box<dyn Future<Output = Result<Value, Error>> + 'a + Send>> + Send,
-    ) -> Result<Value, Error> {
+    ) -> Arc<Result<Value, Error>> {
```

Comment:
Arc changes aren't needed in this PR I think. Unless it gives us improvement in performance.

Reply:
This does provide performance improvement so I think we should keep it